File: 프로그래머스/3/49191. 순위/순위.java
class Solution {
    public int solution(int n, int[][] results) {
        int[][] graph = new int[n+1][n+1];
        for(int[] r : results){
            graph[r[0]][r[1]]= 1;
            graph[r[1]][r[0]]= -1;
        }
        
        for(int k = 1; k <= n; k++){
            for(int i = 1; i <= n; i++){
                for(int j = 1; j <= n; j++){
                    if(graph[i][k] == 1 && graph[k][j] == 1){
                        graph[i][j] = 1;
                    }
                    if(graph[i][k] == -1 && graph[k][j] == -1){
                        graph[i][j] = -1;
                    }
                }
            }
        }
        
        int ans = 0;
        for(int i = 1; i <= n; i++){
            int cnt = 0;
            for(int j = 1; j <= n; j++){
                if(graph[i][j] != 0) cnt++;
            }
            if(cnt == n-1) ans++;
        }
        return ans;
    }
}

File: 프로그래머스/3/49191. 순위/순위.js
function solution(n, results) {
    let score = Array.from({ length : n+1 }, () => Array(n+1).fill(0));
    for(const r of results){
        score[r[0]][r[1]] = 1;
        score[r[1]][r[0]] = -1;
    }
    
    for(let k = 1; k <= n; k++){
        for(let i = 1; i <= n; i++){
            for(let j = 1; j <= n; j++){
                if(score[i][k] === 1 && score[k][j] === 1){
                    score[i][j] = 1;
                }else if(score[i][k] === -1 && score[k][j] === -1){
                    score[i][j] = -1;
                }
            }
        }
    }
    
    let answer = 0;
    for(let i = 1; i <= n; i++){
        let cnt = 0;
        for(let j = 1; j <= n; j++){
            if(score[i][j] != 0){
                cnt++;
            }
        }
        if(cnt === n-1){
            answer++;
        }
    }
    return answer;
}